trace/otel_span.go

```diff
@@ -0,0 +1,110 @@
+package trace
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"strings"
+	"time"
+
+	"google.golang.org/grpc/codes"
+	"go.opentelemetry.io/otel/api/core"
+	oteltrace "go.opentelemetry.io/otel/api/trace"
+)
+
+// implements otel/api/trace.Span. Has methods defined in that interface, but
+// none of them are implemented except for SpanContext. This is just used for
+// compatibilty between beelines and OTel-Go.
+type otelSpan struct {
+	spanContext core.SpanContext
+}
+
+func (os otelSpan) SpanContext() core.SpanContext {
+	return os.spanContext
+}
+
+func (os otelSpan) IsRecording() bool {
+	return false
+}
+
+func (os otelSpan) SetStatus(code codes.Code, msg string) {
+	return
+}
+
+func (os otelSpan) SetAttribute(k string, v interface{}) {
+	return
+}
+
+func (os otelSpan) SetAttributes(attributes ...core.KeyValue) {
+	return
+}
+
+func (os otelSpan) End(options ...oteltrace.EndOption) {
+	return
+}
+
+func (os otelSpan) RecordError(ctx context.Context, err error, opts ...oteltrace.ErrorOption) {
+	return
+}
+
+func (os otelSpan) Tracer() oteltrace.Tracer {
+	return nil
+}
+
+func (os otelSpan) AddEvent(ctx context.Context, name string, attrs ...core.KeyValue) {
+	return
+}
+
+func (os otelSpan) AddEventWithTimestamp(ctx context.Context, timestamp time.Time, name string, attrs ...core.KeyValue) {
+	return
+}
+
+func (os otelSpan) SetName(name string) {
+	return
+}
+
+// OtelSpanFromContext uses the Honeycomb trace and span in the provided
+// context to construct an OpenTelemetry SpanContext and returns it wrapped in
+// an OTel span. This allows the beeline to use certain tracing and propagation
+// methods defined in the OpenTelemetry Go SDK.
+func OtelSpanFromContext(ctx context.Context) oteltrace.Span {
+	sp := GetSpanFromContext(ctx)
+	
+	if sp == nil {
+		return otelSpan{}
+	}
+
+	// OpenTelemetry uses 16 byte trace ids and 8 byte span ids. Because honeycomb
+	// uses uuids for both, we have to truncate span ids before sending them on in
+	// trace headers.
+	traceId, _ := core.TraceIDFromHex(strings.ReplaceAll(sp.GetTrace().GetTraceID(), "-", ""))
+	spanId, _ := core.SpanIDFromHex(strings.ReplaceAll(sp.spanID, "-", "")[:16])
```

Comment:
This will need to be thought out more. It's necessary when sending UUID like strings we generate to Tel, but I'm not comfortable with just an arbitrary chop.